stripe/src/main/java/com/stripe/android/paymentsheet/PaymentSheetFlowControllerFactory.kt

```diff
@@ -0,0 +1,155 @@
+package com.stripe.android.paymentsheet
+
+import android.content.Context
+import com.stripe.android.PaymentConfiguration
+import com.stripe.android.PaymentSessionPrefs
+import com.stripe.android.StripeApiRepository
+import com.stripe.android.model.ListPaymentMethodsParams
+import com.stripe.android.model.PaymentMethod
+import com.stripe.android.networking.ApiRequest
+import com.stripe.android.networking.StripeRepository
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+import kotlin.coroutines.CoroutineContext
+
+internal class PaymentSheetFlowControllerFactory(
+    private val stripeRepository: StripeRepository,
+    private val config: PaymentConfiguration,
+    private val paymentSessionPrefs: PaymentSessionPrefs,
+    private val workContext: CoroutineContext
+) {
+    constructor(
+        context: Context,
+        workContext: CoroutineContext = Dispatchers.IO
+    ) : this(
+        context,
+        PaymentConfiguration.getInstance(context),
+        workContext
+    )
+
+    private constructor(
+        context: Context,
+        config: PaymentConfiguration,
+        workContext: CoroutineContext
+    ) : this(
+        StripeApiRepository(
+            context,
+            config.publishableKey,
+            workContext = workContext
+        ),
+        config,
+        PaymentSessionPrefs.Default(context),
+        workContext
+    )
+
+    fun create(
+        clientSecret: String,
+        ephemeralKey: String,
+        customerId: String,
+        onComplete: (PaymentSheetFlowController) -> Unit
+    ) {
+        CoroutineScope(workContext).launch {
+            dispatchResult(
+                createWithDefaultArgs(clientSecret, ephemeralKey, customerId),
+                onComplete
+            )
+        }
+    }
+
+    fun create(
+        clientSecret: String,
+        onComplete: (PaymentSheetFlowController) -> Unit
+    ) {
+        CoroutineScope(workContext).launch {
+            dispatchResult(
+                createWithGuestArgs(clientSecret),
+                onComplete
+            )
+        }
+    }
+
+    private suspend fun dispatchResult(
+        result: Result,
+        onComplete: (PaymentSheetFlowController) -> Unit
+    ) = withContext(Dispatchers.Main) {
+        when (result) {
+            is Result.Success -> {
+                onComplete(result.flowController)
+            }
+            is Result.Failure -> {
+                // ???
```

Comment:
What's the androidy thing to do here? Throw an exception?

On iOS I'm returning the user this Result.

Reply:
Will change this to return a `Result`